File: src/main/java/com/study/platform/domain/post/dto/response/StudyPostResponse.java
package com.study.platform.domain.post.dto.response;

import com.study.platform.domain.post.model.StudyPost;
import com.study.platform.domain.post.model.StudyPostStatus;
import io.swagger.v3.oas.annotations.media.Schema;

import java.time.LocalDateTime;
import java.util.UUID;

public record StudyPostResponse(

        @Schema(description = "게시글 ID")
        UUID id,

        @Schema(description = "작성자 ID")
        UUID authorId,

        @Schema(description = "작성자 닉네임")
        String authorNickname,

        @Schema(description = "제목")
        String title,

        @Schema(description = "내용")
        String description,

        @Schema(description = "기술스택")
        String techStack,

        @Schema(description = "최대 인원")
        int maxMembers,

        @Schema(description = "모집 마감일")
        LocalDateTime deadline,

        @Schema(description = "모집 상태")
        StudyPostStatus status,

        @Schema(description = "작성일")
        LocalDateTime createdAt,

        @Schema(description = "수정일")
        LocalDateTime updatedAt
) {

    public static StudyPostResponse from(StudyPost post) {
        return new StudyPostResponse(
                post.getId(),
                post.getAuthor().getId(),
                post.getAuthor().getNickname(),
                post.getTitle(),
                post.getDescription(),
                post.getTechStack(),
                post.getMaxMembers(),
                post.getDeadline(),
                post.getStatus(),
                post.getCreatedAt(),
                post.getUpdatedAt()
        );
    }
}

File: application/src/main/kotlin/com/study/platform/domain/post/dto/response/StudyPostResponse.kt
package com.study.platform.domain.post.dto.response

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.study.platform.domain.post.model.StudyPost
import com.study.platform.domain.post.model.StudyPostStatus
import io.swagger.v3.oas.annotations.media.Schema
import java.time.LocalDateTime
import java.util.UUID

data class StudyPostResponse @JsonCreator constructor(

    @JsonProperty("id")
    @field:Schema(description = "게시글 ID")
    val id: UUID,

    @JsonProperty("authorId")
    @field:Schema(description = "작성자 ID")
    val authorId: UUID,

    @JsonProperty("authorNickname")
    @field:Schema(description = "작성자 닉네임")
    val authorNickname: String,

    @JsonProperty("title")
    @field:Schema(description = "제목")
    val title: String,

    @JsonProperty("description")
    @field:Schema(description = "내용")
    val description: String,

    @JsonProperty("techStack")
    @field:Schema(description = "기술스택")
    val techStack: String?,

    @JsonProperty("maxMembers")
    @field:Schema(description = "최대 인원")
    val maxMembers: Int,

    @JsonProperty("deadline")
    @field:Schema(description = "모집 마감일")
    val deadline: LocalDateTime,

    @JsonProperty("status")
    @field:Schema(description = "모집 상태")
    val status: StudyPostStatus,

    @JsonProperty("createdAt")
    @field:Schema(description = "작성일")
    val createdAt: LocalDateTime,

    @JsonProperty("updatedAt")
    @field:Schema(description = "수정일")
    val updatedAt: LocalDateTime
) {
    companion object {
        fun from(post: StudyPost): StudyPostResponse = StudyPostResponse(
            id = post.id!!,
            authorId = post.author!!.id!!,
            authorNickname = post.author!!.nickname,
            title = post.title,
            description = post.description,
            techStack = post.techStack,
            maxMembers = post.maxMembers,
            deadline = post.deadline!!,
            status = post.status,
            createdAt = post.createdAt!!,
            updatedAt = post.updatedAt!!
        )
    }
}
